### PR TITLE
feature(exports) Adding languages to Export Profile

### PR DESCRIPTION
**Pull request by wiewiurdp**
# Description

Adding languages information to Shopware 6 Export Profile

Issue #644 
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit Tests
- [x] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/exporter-shopware-6/features/export-profile-shopware6-api.feature [hunk 1]
```diff
@@ -52,6 +52,24 @@ Feature: Export Profile Shopware 6 API
     Then the response status code should be 201
     And store response param "id" as "attribute_text_id"
 
+  Scenario: Create category tree
+    Given I am Authenticated as "test@ergonode.com"
+    And I add "Content-Type" header equal to "application/json"
+    And I add "Accept" header equal to "application/json"
```
Comment (wfajczyk) [inline module/exporter-shopware-6/features/export-profile-shopware6-api.feature:58]: This data is set in  `Background` section

File: module/exporter-shopware-6/src/Application/Form/Model/ExporterShopware6ConfigurationModel.php [hunk 2]
```diff
@@ -58,6 +58,13 @@ class ExporterShopware6ConfigurationModel
      */
     public ?Language $defaultLanguage = null;
 
+    /**
+     * @var Language[]|null
+     *
+     * @Assert\NotBlank()
```
Comment (wfajczyk) [inline module/exporter-shopware-6/src/Application/Form/Model/ExporterShopware6ConfigurationModel.php:64]: This value is not required

File: module/exporter-shopware-6/src/Domain/Command/CreateShopware6ExportProfileCommand.php [hunk 3]
```diff
@@ -60,6 +60,13 @@ class CreateShopware6ExportProfileCommand implements DomainCommandInterface
      */
     private Language $defaultLanguage;
 
+    /**
+     * @var Language[]
+     *
+     * @JMS\Type("array")
```
Comment (wfajczyk) [inline module/exporter-shopware-6/src/Domain/Command/CreateShopware6ExportProfileCommand.php:66]:
```suggestion
      * @JMS\Type("array<string, Ergonode\Core\Domain\ValueObject\Language>")
```

File: module/exporter-shopware-6/src/Domain/Command/UpdateShopware6ExportProfileCommand.php [hunk 4]
```diff
@@ -59,6 +59,14 @@ class UpdateShopware6ExportProfileCommand implements DomainCommandInterface
      * @JMS\Type("Ergonode\Core\Domain\ValueObject\Language")
      */
     private Language $defaultLanguage;
+
+    /**
+     * @var Language[]
+     *
+     * @JMS\Type("array")
```
Comment (wfajczyk) [inline module/exporter-shopware-6/src/Domain/Command/UpdateShopware6ExportProfileCommand.php:66]:
```suggestion
      * @JMS\Type("array<string, Ergonode\Core\Domain\ValueObject\Language>")
```

File: module/exporter-shopware-6/src/Domain/Command/UpdateShopware6ExportProfileCommand.php [hunk 5]
```diff
@@ -129,6 +137,7 @@ class UpdateShopware6ExportProfileCommand implements DomainCommandInterface
      * @param string              $clientId
      * @param string              $clientKey
      * @param Language            $defaultLanguage
+     * @param array               $languages
```
Comment (wfajczyk) [inline module/exporter-shopware-6/src/Domain/Command/UpdateShopware6ExportProfileCommand.php:140]:
```suggestion
     * @param Language[]               $languages
```

File: module/exporter-shopware-6/src/Domain/Entity/Shopware6ExportApiProfile.php [hunk 6]
```diff
@@ -121,6 +129,7 @@ class Shopware6ExportApiProfile extends AbstractExportProfile implements ExportP
      * @param string              $clientId
      * @param string              $clientKey
      * @param Language            $defaultLanguage
+     * @param array               $languages
```
Comment (wfajczyk) [inline module/exporter-shopware-6/src/Domain/Entity/Shopware6ExportApiProfile.php:132]:
```suggestion
     * @param Language[]               $languages
```

File: module/exporter-shopware-6/src/Domain/Entity/Shopware6ExportApiProfile.php [hunk 7]
```diff
@@ -317,6 +336,14 @@ public function setDefaultLanguage(Language $defaultLanguage): void
         $this->defaultLanguage = $defaultLanguage;
     }
 
+    /**
+     * @param array $languages
```
Comment (wfajczyk) [inline module/exporter-shopware-6/src/Domain/Entity/Shopware6ExportApiProfile.php:340]:
```suggestion
     * @param Language[] $languages
```

File: module/exporter-shopware-6/src/Application/Model/Shopware6ChannelFormModel.php [hunk 8]
```diff
@@ -52,11 +52,16 @@ class Shopware6ChannelFormModel
     public ?string $clientKey = null;
 
     /**
-     * @var Language|null
+     * @var string|null
      *
      * @Assert\NotBlank()
```
Comment (rprzedzik) [inline module/exporter-shopware-6/src/Application/Model/Shopware6ChannelFormModel.php:57]: There is no assertion that language exist and is active

File: module/exporter-shopware-6/src/Application/Model/Shopware6ChannelFormModel.php [hunk 9]
```diff
@@ -52,11 +52,16 @@ class Shopware6ChannelFormModel
...
      */
-    public ?Language $defaultLanguage = null;
+    public ?string $defaultLanguage = null;
+
+    /**
+     * @var array|null
```
Comment (rprzedzik) [inline module/exporter-shopware-6/src/Application/Model/Shopware6ChannelFormModel.php:62]: There is no assertion for collection

File: module/exporter-shopware-6/src/Domain/Entity/Shopware6Channel.php [hunk 10]
```diff
@@ -50,6 +50,14 @@ class Shopware6Channel extends AbstractChannel
      */
     private Language $defaultLanguage;
 
+    /**
+     * @var array
+     *
+     * @JMS\Type("array")
+     *
+     */
+    private array $languages;
```
Comment (rprzedzik) [inline module/exporter-shopware-6/src/Domain/Entity/Shopware6Channel.php:59]: array of what ??

File: module/exporter-shopware-6/src/Domain/Entity/Shopware6Channel.php [hunk 11]
```diff
@@ -308,6 +327,14 @@ public function setDefaultLanguage(Language $defaultLanguage): void
         $this->defaultLanguage = $defaultLanguage;
     }
 
+    /**
+     * @param array $languages
+     */
+    public function setLanguages(array $languages): void
```
Comment (rprzedzik) [inline module/exporter-shopware-6/src/Domain/Entity/Shopware6Channel.php:333]: i can add array of any types its is correct ??

File: module/exporter-shopware-6/src/Domain/Command/CreateShopware6ChannelCommand.php [hunk 12]
```diff
@@ -222,6 +232,14 @@ public function getDefaultLanguage(): Language
         return $this->defaultLanguage;
     }
 
+    /**
+     * @return array
+     */
+    public function getLanguages(): array
```
Comment (rprzedzik) [inline module/exporter-shopware-6/src/Domain/Command/CreateShopware6ChannelCommand.php:238]: Array of ... ?

File: module/exporter-shopware-6/src/Domain/Command/UpdateShopware6ChannelCommand.php [hunk 13]
```diff
@@ -146,6 +154,7 @@ public function __construct(
         string $clientId,
         string $clientKey,
         Language $defaultLanguage,
+        array $languages,
```
Comment (rprzedzik) [inline module/exporter-shopware-6/src/Domain/Command/UpdateShopware6ChannelCommand.php:157]: Array of what ??

**Pull request merged**